### DISCUSSION TITLE
How do I capture multiple siblings with one capture?

### DISCUSSION THREAD

**Discussion by mawkler** [tree-sitter/tree-sitter#2674] [Q&A]
Hi! I'm trying to capture Markdown checkboxes _and_ the minus sign in front of them with as one capture.

Here's my query:

```scm
(
  (list_marker_minus)
  (task_list_marker_unchecked)
) @checkbox
```

However, only the first sibling, `list_marker_minus` gets selected by Treesitter, and `task_list_marker_unchecked` is not included.

Here's a file that I'm using as an example:

```markdown
- [x] foo
- [ ] bar
  - [ ] baz
```

Here's what the syntax tree looks like:

```scm
section [0, 0] - [3, 0]
  list [0, 0] - [3, 0]
    list_item [0, 0] - [1, 0]
      list_marker_minus [0, 0] - [0, 2]
      task_list_marker_checked [0, 2] - [0, 5]
      ...
    list_item [1, 0] - [3, 0]
      list_marker_minus [1, 0] - [1, 2]
      task_list_marker_unchecked [1, 2] - [1, 5]
      ...
      list [2, 2] - [3, 0]
        list_item [2, 2] - [3, 0]
          list_marker_minus [2, 2] - [2, 4]
          task_list_marker_unchecked [2, 4] - [2, 7]
          ...
```

[The documentation](https://tree-sitter.github.io/tree-sitter/using-parsers#grouping-sibling-nodes) says that I can group sibling nodes by wrapping them in parentheses. What am I doing wrong?

I've read through the other discussions but I couldn't find an answer.

**Comment by dabico** [comment]
The following seems to work in the playground for Python:

Query:
```
(
  (identifier) @capture.identifier
  (parameters) @capture.parameters
)
```

Input:
```
def foo(x):
    pass
```

Just change the node and capture names to what you need in markdown.

**Reply by mawkler** [reply]
@dabico That creates two different capture groups. I want one single capture group that captures both node types. The reason is that I'm trying to conceal the dash _and_ checkbox in Neovim with one single character, `\udb80\udd32`, which looks like this with a nerd font (I'm also planning to do the same for unchecked boxes with `\udb80\udd31`):

![image](https://github.com/tree-sitter/tree-sitter/assets/15816726/1dea23a8-1ad8-4f5a-a408-6900dbea6bf1)

In Neovim this is done by creating a treesitter capture and telling Neovim to [conceal that capture](https://neovim.io/doc/user/treesitter.html#treesitter-highlight-conceal) using a `#set! conceal <char>` directive, where `<char>` is the character to replace the capture with.

I therefore need it to be one single capture.

**Reply by dabico** [reply]
Maybe this is not 100% correct (since I can't see the structure of the rest of your tree), but I managed to capture your specific needs in my example by changing the query to the following:

```
(function_definition . (_)* @capture)
```

So yours would swap `function_definition` for  `list_item`.

**Reply by mawkler** [reply]
@dabico I tried your `(_)*` technique but it captures the text after the checkbox as well (`i.e. paragraph`).

This is what the syntax tree looks like btw:

```scm
section [0, 0] - [3, 0]
  list [0, 0] - [3, 0]
    list_item [0, 0] - [1, 0]
      list_marker_minus [0, 0] - [0, 2]
      task_list_marker_checked [0, 2] - [0, 5]
      paragraph [0, 6] - [1, 0]
        inline [0, 6] - [0, 9]
    list_item [1, 0] - [3, 0]
      list_marker_minus [1, 0] - [1, 2]
      task_list_marker_unchecked [1, 2] - [1, 5]
      paragraph [1, 6] - [2, 2]
        inline [1, 6] - [1, 9]
        block_continuation [2, 0] - [2, 2]
      list [2, 2] - [3, 0]
        list_item [2, 2] - [3, 0]
          list_marker_minus [2, 2] - [2, 4]
          task_list_marker_unchecked [2, 4] - [2, 7]
          paragraph [2, 8] - [3, 0]
            inline [2, 8] - [2, 11]
```

**Comment by ahlinc** [comment]
I've replied in: https://github.com/neovim/neovim/issues/25515#issuecomment-1749749109
